internal/pkg/instrumentors/utils/kernel.go

```diff
@@ -15,28 +15,39 @@
 package utils
 
 import (
-	"strings"
+	"fmt"
+	"strconv"
 	"syscall"
 
 	"github.com/hashicorp/go-version"
 )
 
+// parse logic adapted from https://github.com/golang/go/blob/go1.21.3/src/internal/syscall/unix/kernel_version_linux.go
 func GetLinuxKernelVersion() (*version.Version, error) {
-	var utsname syscall.Utsname
-
-	if err := syscall.Uname(&utsname); err != nil {
+	var uname syscall.Utsname
+	if err := syscall.Uname(&uname); err != nil {
 		return nil, err
 	}
 
-	var buf [65]byte
-	for i, v := range utsname.Release {
-		buf[i] = byte(v)
-	}
-
-	ver := string(buf[:])
-	if strings.Contains(ver, "-") {
-		ver = strings.Split(ver, "-")[0]
+	var (
```

Comment:
Was this algorithm taken from somewhere? Can you please cite where it's from if so?

Reply:
already commented. Should I write more details?

- https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/390/files#diff-a79ffed82bd81b27d495d5ab6eee4090890e9d9e8f0df170b97b841703bf2ceeR25